From d366bfbc1ce792e5e1ff411bc6e823644b9e732f Mon Sep 17 00:00:00 2001
From: Shadi Ammouri <shadi@marvell.com>
Date: Thu, 7 Jan 2016 08:46:31 +0200
Subject: [PATCH 0649/1240] env: Add temporary support for NAND based env vars

	Use ifdef to select the env vars location (spi or nand),
	till we add the code that selects the env location depending on
	sample-at-reset.

Change-Id: I524d7fe278496e404e7e0eaee8103033948d2f73
Signed-off-by: Shadi Ammouri <shadi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/26610
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 common/mvebu/env_bootdev.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/common/mvebu/env_bootdev.c b/common/mvebu/env_bootdev.c
index 4b77109..8891cbc 100644
--- a/common/mvebu/env_bootdev.c
+++ b/common/mvebu/env_bootdev.c
@@ -33,7 +33,13 @@ int saveenv(void)
  */
 int env_init(void)
 {
+	/* Use ifdef as a temporary solution till we add sample-at-reset
+	** logic. */
+#ifdef CONFIG_MVEBU_NAND_BOOT
+	nand_env_init();
+#else
 	sf_env_init();
+#endif
 	return gd->arch.env_func.init_env();
 }
 
@@ -42,6 +48,12 @@ void env_relocate_spec(void)
 	/* sf_env_init is called again here because of the
 	 * address relocation, the addreses need to be corrected
 	 */
+	/* Use ifdef as a temporary solution till we add sample-at-reset
+	** logic. */
+#ifdef CONFIG_MVEBU_NAND_BOOT
+	nand_env_init();
+#else
 	sf_env_init();
+#endif
 	gd->arch.env_func.reloc_env();
 }
-- 
1.9.1

